### ISSUE TITLE
Design database model for wishlists

### ISSUE THREAD

**Issue by sakshamgarg** [CSCI-GA-2820-SP23-003/wishlists#9]
**As a** developer
**I need** to design a database model for the wishlists
**So that** all the information regarding wishlists can be stored in a single place and we can apply all the CRUD operations to it.
   
 ### Details and Assumptions
 * We need to come up with the database design. 
 * Need to finalize the tables and the required fields in each table.
   
 ### Acceptance Criteria  
   
 ```gherkin
 Given the application is running
 When any of the CRUD + list operation is performed
 Then that information should be retrieved or saved to the database.
```

**Comment by reubenabraham** [comment]
<img width="1186" alt="image" src="https://user-images.githubusercontent.com/25875077/219971831-e13e2d3e-6b8d-4e93-bf2e-0d85af02604b.png">
Here's how I think it can look. Open to suggestions.

**Comment by sakshamgarg** [comment]
The overall design looks good to me. Though I have couple of points/questions to ask:

1. Why is there an item_id in items table ? Is it majorly to have a primary key in the table ?
2. Should there be more fields in items for each wishlist like 
  a. Priority - This can be a feature to help users have their wishlist sorted according to the priority. Can be Low, Medium or High.
  b. Comment - If there is some comment which a user wants to leave for any item or a wishlist. Can be totally ignored for now maybe?
3. Will there be multiple user profiles in this e-commerce website ? If yes, then can there be multiple users accessing the same wishlist ?

**Comment by reubenabraham** [comment]
Hey Saksham, let me clarify -
1. Yes, it's to uniquely identify a row in the `items` table. Although technically speaking, a combination of (wishlist_id and product_id) can also uniquely identify a row, and can serve as the PK for `items`.
2. As discussed over our call, we can consider these feature enhancements once we have the required functionality up and running first. I like the ideas!
3. As per my understanding, there will be multiple distinct users, but only a **single** user profile per user. Each user should be uniquely identifiable by their `user_id` alone. @jjrofrano - prof- please correct me if this is wrong.

**Comment by reubenabraham** [comment]
Here is the updated schema design for the `wishlists` table(s). Closing this ticket because we have reached a consensus among our team members regarding this.
- Added the extra `product_name` field just so we can return a better looking response to the GET end-point calls, as opposed to returning just a `product_id`.

![image.png](https://images.zenhubusercontent.com/63e6c66e459d4a0a3f841ddc/7be98e6f-4467-484a-ab21-1a343b2763ab)
